src/protocol/IERC20Bridge.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+/// @dev Bridges ERC20 tokens by creating and verifying ERC20Deposits.
+///
+/// These can be created by calling the deposit function. Later, the receiver can
+/// claim the deposit on the destination chain by using a storage proof.
+interface IERC20Bridge {
+    struct TokenInitialization {
+        // The original token address on the source chain
+        address originalToken;
+        // The token name
+        string name;
+        // The token symbol
+        string symbol;
+        // The token decimals
+        uint8 decimals;
+    }
+
+    struct ERC20Deposit {
+        // The nonce of the deposit
+        uint256 nonce;
+        // The sender of the deposit
+        address from;
+        // The receiver of the deposit
+        address to;
+        // The original ERC20 token address (always refers to the original token, not bridged)
+        address localToken;
+        // The amount of the deposit
+        uint256 amount;
+    }
+
+    /// @dev Emitted when a token is initialized on the source chain.
+    /// @param id The initialization id
+    /// @param initialization The token initialization data
+    event TokenInitialized(bytes32 indexed id, TokenInitialization initialization);
+
+    /// @dev Emitted when a token initialization is proven on the destination chain.
+    /// @param id The initialization id
+    /// @param initialization The token initialization data
+    /// @param deployedToken The address of the deployed bridged token
+    event TokenInitializationProven(
+        bytes32 indexed id, TokenInitialization initialization, address indexed deployedToken
+    );
+
+    /// @dev Emitted when a deposit is made.
+    /// @param id The deposit id
+    /// @param deposit The ERC20 deposit
+    event DepositMade(bytes32 indexed id, ERC20Deposit deposit);
+
+    /// @dev Emitted when a deposit is claimed.
+    /// @param id The deposit id
+    /// @param deposit The claimed ERC20 deposit
+    event DepositClaimed(bytes32 indexed id, ERC20Deposit deposit);
+
+    /// @dev Failed to claim the deposit.
+    error FailedClaim();
```

Comment:
I think this is an unused error

Reply:
indeed, removed https://github.com/OpenZeppelin/minimal-rollup/pull/146/commits/cfaec5808bf26a4d841bc95503ecda8f17a94483